### ISSUE TITLE
Can not change machine type of node-pool

### ISSUE THREAD

**Issue by stayanti** [terraform-google-modules/terraform-google-kubernetes-engine#1662]
### TL;DR

When I change the node-machine type of a node-pool, the plan will throw permission exceptions to get the namespace and k8s service accounts, while im sure the user has the correct rights

### Expected behavior

i expected a node pool destroy and recreate of the nodepool

### Observed behavior

terraform plan threw 
│ Error: serviceaccounts "xxx" is forbidden: User "xxxy@xxx" cannot get resource "serviceaccounts" in API group "" in the namespace "xxx": requires one of ["container.serviceAccounts.get"] permission(s).
│ 
│   with kubernetes_service_account.ui,
│   on component-ui.tf line 12, in resource "kubernetes_service_account" "ui":
│   12: resource "kubernetes_service_account" "ui" {
│ 
╵
╷
│ Error: namespaces "message-gateway" is forbidden: User "xxx@xxx" cannot get resource "namespaces" in API group "" in the namespace "xxx": requires one of ["container.namespaces.get"] permission(s).
│ 
│   with kubernetes_namespace.default_application_namespace,
│   on kubernetes.tf line 77, in resource "kubernetes_namespace" "default_application_namespace":
│   77: resource "kubernetes_namespace" "default_application_namespace" {
│ 
╵

### Terraform Configuration

```hcl
module "gke" {
  source                            = "terraform-google-modules/kubernetes-engine/google//modules/private-cluster"
  version                           = "v24.1.0"
  project_id                        = var.project_id
  name                              = local.cloud_resource_name
  regional                          = var.kubernetes_regional_cluster
  region                            = var.region
  zones                             = var.kubernetes_zones
  release_channel                   = var.kubernetes_release_channel
  kubernetes_version                = "latest"
  network                           = module.gcp-network.network_name
  subnetwork                        = module.gcp-network.subnets_names[0]
  ip_range_pods                     = var.ip_range_pods_name
  ip_range_services                 = var.ip_range_services_name
  disable_legacy_metadata_endpoints = true
  cluster_autoscaling               = var.kubernetes_cluster_autoscaling
  enable_private_endpoint           = false
  enable_private_nodes              = true
  master_authorized_networks        = var.kubernetes_management_firewall_allowlist
  create_service_account            = false
  remove_default_node_pool          = true
  http_load_balancing               = true
  horizontal_pod_autoscaling        = false

  node_pools = [
    {
      name               = "pool-01"
      machine_type       = var.kubernetes_node_type
      image_type         = "COS_CONTAINERD"
      enable_gvnic       = true
      spot               = var.kubernetes_use_spot_vm
      kubernetes_version = "latest"
      release_channel    = var.kubernetes_release_channel
      disk_size_gb       = 50
      node_locations     = var.kubernetes_zones[0]
      disk_type          = "pd-standard"
      service_account    = google_service_account.kubernetes_sa.email
      auto_upgrade       = true
      initial_node_count = var.kubernetes_initial_node_count
      min_count          = var.kubernetes_node_min_count
      max_count          = var.kubernetes_node_max_count
    }
  ]

  node_pools_oauth_scopes = {
    all = [
      "https://www.googleapis.com/auth/logging.write",
      "https://www.googleapis.com/auth/monitoring",
      "https://www.googleapis.com/auth/cloud-platform",
    ]
  }

  node_pools_labels = {
    all = {}
  }

  node_pools_metadata = {
    all = {}
  }

  node_pools_taints = {
    all = []
  }

  node_pools_tags = {
    all = []
  }
  depends_on = [
    google_project_service.default
  ]
}

resource "kubernetes_namespace" "default_application_namespace" {
  metadata {
    name = var.application_name
  }
  depends_on = [
    module.gke
  ]
}
```

### Terraform Version

```sh
provider "registry.terraform.io/hashicorp/google" {
  version     = "4.56.0"
```

### Additional information

same problem occurs when i change the name of the node-pool.

**Comment by thamurepo** [comment]
@stayanti  could you pls share your TF files, [ variable.tf] pls.  

or can you pls share what u declare variable for "**var.kubernetes_release_channel"**
2. under module  -   version                           = "v24.1.0"  referring what
